movement/watch_faces/complication/alarm_face.c

```diff
@@ -0,0 +1,414 @@
+    if (state->alarm_handled_minute == now.unit.minute) return false;
+    state->alarm_handled_minute = now.unit.minute;
+    // check the rest
+    for (uint8_t i = 0; i < ALARM_ALARMS; i++) {
+        if (state->alarm[i].enabled) {
+            if (state->alarm[i].minute == now.unit.minute) {
+                if (state->alarm[i].hour == now.unit.hour) {
+                    state->alarm_playing_idx = i;
+                    if (state->alarm[i].day == ALARM_DAY_EACH_DAY) return true;
+                    if (state->alarm[i].day == ALARM_DAY_ONE_TIME) {
+                        // erase this alarm
+                        state->alarm[i].day = ALARM_DAY_EACH_DAY;
+                        state->alarm[i].minute = state->alarm[i].hour = 0;
+                        state->alarm[i].enabled = false;
+                        _alarm_update_alarm_enabled(settings, state);
+                        return true;
+                    }
+                    uint8_t weekday_idx = _get_weekday_idx(now);
+                    if (state->alarm[i].day == weekday_idx) return true;
+                    if (state->alarm[i].day == ALARM_DAY_WORKDAY && weekday_idx < 5) return true;
+                    if (state->alarm[i].day == ALARM_DAY_WEEKEND && weekday_idx >= 5) return true;
+                }
+            }
+        }
+    }
+    state->alarm_handled_minute = -1;
+    return false;
+}
+
+bool alarm_face_loop(movement_event_t event, movement_settings_t *settings, void *context) {
+    (void) settings;
+    alarm_state_t *state = (alarm_state_t *)context;
+
+    switch (event.event_type) {
+    case EVENT_TICK:
+        if (state->alarm_quick_ticks >= 0) {
+            // we are counting ticks for the alarm button
+            if (state->setting_state == alarm_setting_idx_hour || state->setting_state == alarm_setting_idx_minute) {
+                if (state->alarm_quick_ticks < INT8_MAX) state->alarm_quick_ticks++;
+                if (state->alarm_quick_ticks == ALARM_QUICK_MIN_TICKS) {
+                    // initiate fast counting
+                    movement_request_tick_frequency(8);
+                } else if (state->alarm_quick_ticks > ALARM_QUICK_MIN_TICKS) {
+                    // fast count hours or minutes
+                    if (state->setting_state == alarm_setting_idx_hour)
+                        state->alarm[state->alarm_idx].hour = (state->alarm[state->alarm_idx].hour + 1) % 24;
+                    else if (state->setting_state == alarm_setting_idx_minute)
+                        state->alarm[state->alarm_idx].minute = (state->alarm[state->alarm_idx].minute + 1) % 60;
+                    _alarm_face_draw(settings, state, event.subsecond);
+                    break;
+                }
+            } else {
+                _abort_quick_ticks(state);
+            }
+        } else if (!state->is_setting) break; // no need to do anything when we are not in settings mode and no quick ticks are running
+        // otherwise fall through
```

Comment:
Minor tweak: if you remove the "otherwise" (so the comment reads exactly `// fall through`), it will silence a gcc compiler warning that's currently showing up.

Reply:
My compiler does not throw a warning there, but thanks for the hint. Will correct this.